log-manager/src/main/java/io/airlift/log/RollingFileHandler.java

```diff
@@ -92,74 +234,352 @@ public void publish(LogRecord record)
         }
 
         try {
-            asyncAppender.doAppend(message);
+            putUninterruptibly(queue, message);
         }
         catch (Exception e) {
             // catch any exception to assure logging always works
             reportError(null, e, WRITE_FAILURE);
         }
+
+        // if closed while queueing, try to remove the message just to clean up
+        if (closed.get()) {
+            queue.remove(message);
```

Comment:
`queue.clear()` ?

Reply:
Can't clear the whole queue here, there could be messages in the queue that still need to be logged.